apps/settings/js/bluetooth.js

```diff
@@ -685,18 +688,25 @@ navigator.mozL10n.once(function bluetoothSettings() {
     }
 
     function setDeviceUnpair(device) {
+      function unpairDevice() {
+        // backend takes responsibility to disconnect first.
+        var req = defaultAdapter.unpair(device.address);
+        req.onerror = function bt_pairError() {
+          showDevicePaired(true, null);
+        };
+      }
+
       if (device.address === connectedAddress) {
-        var msg = _('unpair-title') + '\n' + _('unpair-msg');
-        if (!window.confirm(msg)) {
-          return;
-        }
-        connectedAddress = null;
+        navigator.mozL10n.formatValue('unpair-confirm').then(msg => {
+          if (!window.confirm(msg)) {
+            return;
+          }
```

Comment:
I'm not pretty sure if it would cause any side effect of being asynchronized. Imagine at the moment user clicking 'unpair' and the UI wasn't frozen by this confirm window *immediately* so the user can click others.. and then at some point the window pops up which may be unexpected. Do you think the promise will be resovled very fast and almost immediately?